homeassistant/components/media_player/sonos.py

```diff
@@ -38,12 +38,23 @@
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """ Sets up the Sonos platform. """
     import soco
+    import socket
 
     if discovery_info:
         add_devices([SonosDevice(hass, soco.SoCo(discovery_info))])
         return True
 
-    players = soco.discover()
+    players = None
+    hosts = config.get('hosts', None)
+    if hosts:
+        players = []
+        for host in hosts.split(","):
+            host = socket.gethostbyname(host)
```

Comment:
In my setup i use dhcp so the ip changes depending on the dhcp daemon
settings, but the hostname is the same.

This works with both hostbames and ips.

/T

On Thu, Jan 21, 2016, 04:23 Paulus Schoutsen <notifications@github.com>
wrote:

> In homeassistant/components/media_player/sonos.py
> <https://github.com/balloob/home-assistant/pull/937#discussion_r50356352>:
>
> >
> >      if discovery_info:
> >          add_devices([SonosDevice(hass, soco.SoCo(discovery_info))])
> >          return True
> >
> > -    players = soco.discover()
> > +    players = None
> > +    hosts = config.get('hosts', None)
> > +    if hosts:
> > +        players = []
> > +        for host in hosts.split(","):
> > +            host = socket.gethostbyname(host)
>
> What would be the added value of getting the host? Doesn't SoCo work with
> just IP addresses?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/balloob/home-assistant/pull/937/files#r50356352>.
>